Branch-SDK/Branch-SDK/Requests/BranchOpenRequest.m

```diff
@@ -109,6 +109,29 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     [BNCSystemObserver setUpdateState];
 
     NSString *sessionData = data[BRANCH_RESPONSE_KEY_SESSION_DATA];
+    if (sessionData == nil || [sessionData isKindOfClass:[NSString class]]) {
+    } else
+    if (([sessionData isKindOfClass:[NSDictionary class]] ||
+         [sessionData isKindOfClass:[NSArray class]]) &&
+            [NSJSONSerialization isValidJSONObject:sessionData]) {
```

Comment:
Should this code reside in BNCEncodingUtils similar to https://github.com/BranchMetrics/ios-branch-deep-linking/pull/674/files#diff-4a8bffd322f48b39ef8419977f64ae8bR146?